subprojects/docs/src/docs/userguide/dependencyManagementTerminology.adoc

```diff
@@ -0,0 +1,44 @@
+[[dependency_management_terminology]]
+== Dependency Management Terminology
+
+Dependency management comes with a wealth of terminology. Here you can find the most commonly-used terms including references to the user guide to learn about their practical application.
+
+[[sub:terminology_module]]
+Module::
+
+A piece of software that evolves over time e.g. link:https://github.com/google/guava[Google Guava]. Every module has a name. Each release of a module is optimally represented by a <<sub:terminology_version,version>>. For convenient consumption, modules can be hosted in a <<sub:terminology_repository,repository>>.
+
+[[sub:terminology_version]]
+Version::
+
+A <<sub:terminology_module,module>> indicates its revision with the help of a version. Each module version contains a distinct set of changes. In practice there's no limitation to the scheme of the version. Timestamps, numbers, special suffixes like GA are all allowed identifiers. The most widely-used versioning strategy is link:https://semver.org/[semantic versioning]. Semantic versioning is represented by the scheme _major_._minor_._patch_ e.g. `1.4.15`. Each part conveys a special meaning about the backwards-compatibility of the release.
+
+[[sub:terminology_dependency]]
+Dependency::
+
+A dependency is defined as a versioned or unversioned module. Dependencies may provide <<sub:terminology_dependency_metadata,metadata>> to indicate that they depend on other dependencies required for it work properly, so-called <<sub:terminology_transitive_dependency,transitive dependencies>>. For more information, see <<declaring_dependencies>>. Gradle's dependency resolution process can apply constraints declared in the build script to select a specific version.
+
+[[sub:terminology_transitive_dependency]]
+Transitive dependency::
+
```

Comment:
A transitive dependency requirement of A is a dependency requirement of `B v1.4`, where `B v1.4` is a resolved dependency of A.
So if A has a dependency requirement on `B [1.0, 2.0)`, which is then resolved to `B v1.4`, and `B v1.4` has a dependency requirement on `C 1.0`, then A has a transitive dependency requirement on `C 1.0`.

Note that I wouldn't bother referring to `C:1.0` as a "transitive resolved dependency": it's just a "resolved dependency" that is brought in by a "transitive dependency requirement".